docs/content/guides/getting-started/binding-to-data.md

```diff
@@ -718,20 +670,127 @@ To change the number of rows or columns rendered by default, use the [`startRows
+handling data aren't prepared for external changes that are made in this way.
+:::
+
+To avoid this scenario, copy the data before you pass it to the grid. To change the data from outside Handsontable, you
+can use our API methods. For example, a change being made will be displayed immediately on the screen after calling
+the [`setDataAtCell()`](@/api/core.md#setdataatcell) method.
+
+::: only-for javascript
+::: example #example10
+
+```js
+const container = document.getElementById('example10');
+
+const data = [
+  ['', 'Tesla', 'Nissan', 'Toyota', 'Honda', 'Mazda', 'Ford'],
+  ['2017', 10, 11, 12, 13, 15, 16],
+  ['2018', 10, 11, 12, 13, 15, 16],
+  ['2019', 10, 11, 12, 13, 15, 16],
+  ['2020', 10, 11, 12, 13, 15, 16],
+  ['2021', 10, 11, 12, 13, 15, 16]
+];
+const settings = {
+  data: data,
+  height: 'auto',
+  licenseKey: 'non-commercial-and-evaluation'
+};
+
+const hot = new Handsontable(container, settings);
+
+hot.setDataAtCell(0, 1, 'Ford');
+```
+
+:::
+:::
+
+::: only-for react
+::: example #example10 :react
+
+```jsx
+import { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { HotTable } from '@handsontable/react';
+import { registerAllModules } from 'handsontable/registry';
+
+// register Handsontable's modules
+registerAllModules();
+
+const ExampleComponent = () => {
+  const hotRef = React.createRef();
+
+  useEffect(() => {
+    const hot = hotRef.current.hotInstance;
+
+    hot.setDataAtCell(0, 1, 'Ford');
+  });
+
+  return (
+      <HotTable 
+        ref={hotRef}
+        data={[
+          ['', 'Tesla', 'Nissan', 'Toyota', 'Honda', 'Mazda', 'Ford'],
+          ['2017', 10, 11, 12, 13, 15, 16],
+          ['2018', 10, 11, 12, 13, 15, 16],
+          ['2019', 10, 11, 12, 13, 15, 16],
+          ['2020', 10, 11, 12, 13, 15, 16],
+          ['2021', 10, 11, 12, 13, 15, 16]
+        ]}
+        height="auto"
+        licenseKey="non-commercial-and-evaluation"
+      />
+  );
+};
+
+ReactDOM.render(<ExampleComponent />, document.getElementById('example10'));
+
+```
+
+:::
+:::
+
 There are multiple ways you can insert your data into Handsontable. Let's go through the most useful ones:
 
 ### The [`data`](@/api/options.md#data) configuration option
+::: only-for javascript
 You will probably want to initialize the table with some data (if you don't, the table will render an empty 5x5 grid for you). The easiest way to do it is passing your data array as [`data`](@/api/options.md#data) option in the initial config object:
 ```js
 const hot = new Handsontable(container, {
   data: newDataset,
   // ... other config options
 });
 ```
+:::
+
+::: only-for react
+You will probably want to initialize the table with some data (if you don't, the table will render an empty 5x5 grid for
+you). The easiest way to do it is by passing your data array as the value of `HotTable`'s [`data`](@/api/options.md#data) prop:
+```jsx
+<HotTable data={dataArray} />
```

Comment:
Please avoid introducing discrepancies between the snippets. 

Is there a reason why the variable name `newDataset` fits better in JS and `dataArray` fits better in React? If not, then please decide on one name in both snippets so we don't have to worry about that.

Reply:
Will change